apps/native/app/src/screens/license-scanner/license-scanner.tsx

```diff
@@ -90,127 +115,143 @@ export const LicenseScannerScreen: NavigationFunctionComponent = ({
 }) => {
   useNavigationOptions(componentId)
   useConnectivityIndicator({ componentId, rightButtons: RIGHT_BUTTONS })
-  const [hasPermission, setHasPermission] = useState<boolean>()
-  const [active, setActive] = useState(true)
-  const [invalid, setInvalid] = useState<boolean>(false)
-  const [torch, setTorch] = useState<boolean>(false)
-
-  const camera = useRef<Camera>()
-  const [layout, setLayout] = useState<LayoutRectangle>()
-  const [ratio, setRatio] = useState<string>()
-  const [padding, setPadding] = useState(0)
-
-  const invalidTimeout = useRef<ReturnType<typeof setTimeout>>()
   const intl = useIntl()
 
-  const [verifyLicenseBarcode, { data, error, loading }] =
-    useVerifyLicenseBarcodeMutation()
+  const [active, setActive] = useState(true)
+  const [invalid, setInvalid] = useState(false)
+  const [torch, setTorch] = useState(false)
 
-  useEffect(() => {
-    Camera.requestCameraPermissionsAsync().then(({ status }) => {
-      setHasPermission(status === 'granted')
-    })
-  }, [])
+  const scanningRef = useRef<ScanningStatus>(ScanningStatus.INITIAL)
+  const { hasPermission, requestPermission } = useCameraPermission()
+  const device = useCameraDevice('back')
+  const zoom = useSharedValue(device?.neutralZoom || 1)
+  const zoomOffset = useSharedValue(0)
 
-  useEffect(() => {
-    Navigation.events().registerNavigationButtonPressedListener((e) => {
-      if (e.buttonId === 'LICENSE_SCANNER_DONE') {
-        Navigation.dismissModal(componentId)
-      }
+  const gesture = Gesture.Pinch()
+    .onBegin(() => {
+      zoomOffset.value = zoom.value
     })
-    Navigation.events().registerComponentWillAppearListener((e) => {
-      if (e.componentId === componentId) {
-        setActive(true)
-      }
+    .onUpdate((event) => {
+      const z = zoomOffset.value * event.scale
+      zoom.value = interpolate(
+        z,
+        [1, 10],
+        [device?.minZoom ?? 1, device?.maxZoom ?? 2],
+        Extrapolation.CLAMP,
+      )
     })
-  }, [])
 
-  const onFlashlightPress = useCallback(() => {
-    setTorch((v) => !v)
-  }, [])
+  const animatedZoom = useAnimatedProps<CameraProps>(
+    () => ({ zoom: zoom.value }),
+    [zoom],
+  )
+
+  const onCodeScanned: CodeScanner['onCodeScanned'] = (codes) => {
+    const { type, value } = codes?.[0]
 
-  const onBarCodeScanned = useCallback(({ type, data }: BarCodeEvent) => {
-    if (invalidTimeout.current) {
-      clearTimeout(invalidTimeout.current)
+    if (!type || !value || scanningRef.current !== ScanningStatus.INITIAL) {
```

Comment:
Currently now in production there are multiple requests made for barcode verification since the scanner executes very frequently. This check prevents that